script/workflowDeploymentAndSetupScript/DeployAndSetupQueueBasedPIM.s.sol

```diff
@@ -0,0 +1,862 @@
+
+        _buyFeeInBps = vm.envUint("BUY_FEE_IN_BPS");
+        // require(_buyFeeInBps > 0, "Buy fee not set");
+
+        _maxBuyFeeInBps = vm.envUint("MAX_BUY_FEE_IN_BPS");
+        require(_maxBuyFeeInBps > 0, "Max buy fee not set");
+
+        _sellFeeInBps = vm.envUint("SELL_FEE_IN_BPS");
+        // require(_sellFeeInBps > 0, "Sell fee not set");
+
+        _maxSellFeeInBps = vm.envUint("MAX_SELL_FEE_IN_BPS");
+        require(_maxSellFeeInBps > 0, "Max sell fee not set");
+
+        _isDirectOperationOnly = vm.envBool("IS_DIRECT_OPERATION_ONLY_BOOL");
+
+        _whitelistRole = vm.envAddress("WHITELIST_ROLE_ADDRESS");
+        require(_whitelistRole != address(0), "Whitelist role not set");
+
+        _whitelistRoleAdmin = vm.envAddress("WHITELIST_ROLE_ADMIN_ADDRESS");
+        require(
+            _whitelistRoleAdmin != address(0), "Whitelist role admin not set"
+        );
+
+        _queueExecutorRole = vm.envAddress("QUEUE_EXECUTOR_ROLE_ADDRESS");
+        require(_queueExecutorRole != address(0), "Queue executor role not set");
+
+        _queueExecutorRoleAdmin =
+            vm.envAddress("QUEUE_EXECUTOR_ROLE_ADMIN_ADDRESS");
+        require(
+            _queueExecutorRoleAdmin != address(0),
+            "Queue executor role admin not set"
+        );
+
+        // Oracle
+        // ---------------------------------------------------------------------
+        _priceSetterRole = vm.envAddress("PRICE_SETTER_ROLE_ADDRESS");
+        require(_priceSetterRole != address(0), "Price setter role not set");
+
+        _priceSetterRoleAdmin = vm.envAddress("PRICE_SETTER_ROLE_ADMIN_ADDRESS");
+        require(
+            _priceSetterRoleAdmin != address(0),
+            "Price setter role admin not set"
+        );
+
+        // Manual Queue Payment Processor
+        // ---------------------------------------------------------------------
+        _cancelledOrdersTreasury =
+            vm.envAddress("CANCELLED_ORDER_TREASURY_ADDRESS");
+        require(
+            _cancelledOrdersTreasury != address(0),
+            "Cancelled orders treasury not set"
+        );
+
+        _failedOrdersTreasury = vm.envAddress("FAILED_ORDER_TREASURY_ADDRESS");
+        require(
+            _failedOrdersTreasury != address(0),
+            "Failed orders treasury not set"
+        );
+
+        _queueOperatorRole = vm.envAddress("QUEUE_OPERATOR_ROLE_ADDRESS");
+        require(_queueOperatorRole != address(0), "Queue operator role not set");
+
+        _queueOperatorRoleAdmin =
+            vm.envAddress("QUEUE_EXECUTOR_ROLE_ADMIN_ADDRESS");
+        require(
+            _queueOperatorRoleAdmin != address(0),
+            "Queue operator role admin not set"
+        );
+
+        // Workflow config
+        // ---------------------------------------------------------------------
+        _independentUpdateBool = vm.envBool("INDEPENDENT_UPDATE_BOOL");
+        if (_independentUpdateBool) {
+            _independentUpdateAdmin =
```

Comment:
there are definitely cases where someone might want to have this set to 0 while having the update bool set to `true` (un-updatable workflows), but you may still (for this usecase) want to check that the admin (if set, like in this branch) is not the zero address, as in their case they probably do not want that (i assume).

Reply:
They don't want independent updates afaik. I will remove the if statement ser